src/app/store/test-records/reducers/test-records.reducer.ts

```diff
@@ -73,11 +76,56 @@ export const testResultsReducer = createReducer(
   on(initialContingencyTest, (state, action) => ({
     ...state,
     editingTestResult: { ...action.testResult } as TestResultModel
-  }))
+  })),
+  on(createDefect, (state, action) => ({ ...state, editingTestResult: createNewDefect(state.editingTestResult, action.defect) })),
+  on(updateDefect, (state, action) => ({ ...state, editingTestResult: updateDefectAtIndex(state.editingTestResult, action.defect, action.index) })),
+  on(removeDefect, (state, action) => ({ ...state, editingTestResult: removeDefectAtIndex(state.editingTestResult, action.index) }))
 );
 
 export const testResultsFeatureState = createFeatureSelector<TestResultsState>(STORE_FEATURE_TEST_RESULTS_KEY);
 
+function createNewDefect(testResultState: TestResultModel | undefined, defect: TestResultDefect): TestResultModel | undefined {
+  if (!testResultState) {
+    return;
+  }
+  const testResult = cloneDeep(testResultState);
+  const testType = testResult.testTypes[0];
+  if (!testType.defects) {
+    return;
+  }
+  testType.defects.push(defect);
+
+  return { ...testResult };
```

Comment:
Here you are instantiating a `testType` variable and assigning it a value, but then you return `testResult` without using `testType`.

Are you trying to modify the value of `testResult.testTypes[0]` by reference?

Reply:
If the answer to that is yes, modifying something by reference is bad practice and should be avoided, unless there is no other way of achieving the same outcome.

In line 98, 112 & 126 `return { ...testResult, testType }`.